composer/datasets/lm_datasets.py

```diff
@@ -85,13 +89,36 @@ def initialize_object(self, batch_size: int, dataloader_hparams: DataloaderHpara
                 if using pip or `conda install -c conda-forge datasets transformers` if using Anaconda.""")) from e
 
         self.validate()
-        self.tokenizer = transformers.AutoTokenizer.from_pretrained(self.tokenizer_name)  #type: ignore (thirdparty)
-        self.config = transformers.AutoConfig.from_pretrained(self.tokenizer_name)  #type: ignore (thirdparty)
-        lm_datasets = [datasets.load_from_disk(i) for i in self.datadir]  #type: ignore (thirdparty)
+        if self.use_synthetic:
+            column_names = ["text"]
+
+            # we just use the max sequence length in tokens to upper bound the sequence length in characters
+            lm_datasets = SyntheticHFDataset(num_samples=self.synthetic_num_unique_samples,
+                                             chars_per_sample=self.train_sequence_length,
+                                             column_names=column_names)
+            lm_datasets = lm_datasets.generate_dataset()
+
+            # flatten the columnar dataset into one column
+            flattened_dataset = functools.reduce(operator.iconcat, [lm_datasets[i] for i in column_names], [])
+            self.tokenizer = SyntheticBertTokenizer(flattened_dataset)
+
+            lm_datasets = lm_datasets.map(lambda inp: self.tokenizer(
+                text=inp[column_names[0]
```

Comment:
this won't handle column_names > 1? or is the (unenforced) convention that `column_names[0]` should always be the text?